server/middlewares/authentication.coffee

```diff
@@ -19,14 +21,50 @@ module.exports.authenticate = (req, res, next) ->
                 if err
                     next new Error localization.t 'error keys not intialized'
                 else
-                    req.logIn user, (err, info) ->
-                        if err
-                            msg = localization.t "error login failed"
-                            error = new Error msg
-                            error.status = 401
-                            next error
-                        else
-                            res.status(200).send success: true
+                    postLogin user
+
+    # Handling login error (if there's one), and dispatching the process
+    # to the right 2FA method (none, HOTP or TOTP)
+    postLogin = (user) ->
+        otpManager.getAuthType (err, otpAuth) ->
+            if err
+                msg = localization.t "error login failed"
+                error = new Error msg
+                error.status = 401
+                next error
+            else if not otpAuth
+                req.logIn user, (err, info) ->
+                    if err
+                        msg = localization.t "error login failed"
+                        error = new Error msg
+                        error.status = 401
+                        next error
+                    else
+                        res.status(200).send success: true
+            else
+                passport.authenticate(otpAuth, processOtp)(req, res, next)
+
+    processOtp = (err, user) ->
+        if err
+            msg = localization.t err
+            error = new Error msg
+            next error
+        else if not user and user isnt undefined
```

Comment:
I'm wondering if, in this case, `unless` scopes `user or user is undefined` or just `user`